pos_frequency_pipeline.py

```diff
@@ -10,16 +17,48 @@ class EmptyFileError(Exception):
 
 
 class POSFrequencyPipeline:
-    def __init__(self, corpus):
-        pass
+    def __init__(self, corpus_manager):
+        self.corpus_manager = corpus_manager
 
     def run(self):
-        pass
+        for article in self.corpus_manager.get_articles().values():
```

Comment:
split the method into several according to SRP. how do you think what separate things are done in one place in your implementation?